### ISSUE TITLE
not support MSVC

### ISSUE THREAD

**Issue by tkorays** [Snaipe/libcsptr#26]
We use C in windows with MSVC compiler, but `__attribute__((cleanup))` does not work in MSVC. Is is possible to support MSVC?

**Comment by Snaipe** [comment]
Not really. MSVC has no practical equivalent to the cleanup attribute, which is why there is a disclaimer that this only works for GNU C.

**Comment by bogen85** [comment]
Clang and Intel C Compiler supposedly support this.
https://clang.llvm.org/docs/AttributeReference.html#cleanup
https://community.intel.com/t5/Intel-C-Compiler/variable-attribute-cleanup-support/td-p/777276

**Comment by Snaipe** [comment]
Yes, clang and icc count as a GNU-flavored C, since they want to match GCC's general behavior. But MSVC does not, and therefore cannot work with this library.

**Comment by malkia** [comment]
I wonder if MSVC's only "__try" / "__finally" can be used - although that's probably not good idea, but I was looking for RAII methods in "C" and just found this.
